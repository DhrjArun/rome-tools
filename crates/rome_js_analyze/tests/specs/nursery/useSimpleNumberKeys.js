// invalid

({ 1n: 1 });
({ 0x1: 1 });
({ 012: 1 });
({ 0b1: 1 });
({ 0o1: 1 });
({ 1_0: 1 });
({ 0.1e1_2: "ee" });
({ 11_1.11: "ee" });

// valid

({ 0: "zero" });
({ 1: "one" });
({ 1.2: "12" });
({ 3.1e12: "12" });
({ 0.1e12: "ee" });
