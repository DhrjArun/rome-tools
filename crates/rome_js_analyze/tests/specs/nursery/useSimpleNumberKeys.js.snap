---
source: crates/rome_js_analyze/tests/spec_tests.rs
expression: useSimpleNumberKeys.js
---
# Input
```js
// valid
({ 0: "zero" });
({ 1: "one" });
({ 1.2: "12" });
({ 3.1e12: "12" });
({ 0.1e12: "ee" });

// invalid
({ 1n: 1 });
({ 0x1: 1 });
({ 012: 1 });
({ 0b1: 1 });
({ 0o1: 1 });
({ 1_0: 1 });
({ 0.1e1_2: "ed" });
({ 11_1.11: "ee" });

```

# Diagnostics
```
useSimpleNumberKeys.js:9:4 lint/nursery/useSimpleNumberKeys  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Bigint is not allowed here.
  
     8 │ // invalid
   > 9 │ ({ 1n: 1 });
       │    ^^
    10 │ ({ 0x1: 1 });
    11 │ ({ 012: 1 });
  
  i Safe fix: Replace 1n with 1
  
     7  7 │   
     8  8 │   // invalid
     9    │ - ({·1n:·1·});
        9 │ + ({·1:·1·});
    10 10 │   ({ 0x1: 1 });
    11 11 │   ({ 012: 1 });
  

```

```
useSimpleNumberKeys.js:10:4 lint/nursery/useSimpleNumberKeys  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Hexadecimal number literal is not allowed here.
  
     8 │ // invalid
     9 │ ({ 1n: 1 });
  > 10 │ ({ 0x1: 1 });
       │    ^^^
    11 │ ({ 012: 1 });
    12 │ ({ 0b1: 1 });
  
  i Safe fix: Replace 0x1 with 1
  
     8  8 │   // invalid
     9  9 │   ({ 1n: 1 });
    10    │ - ({·0x1:·1·});
       10 │ + ({·1:·1·});
    11 11 │   ({ 012: 1 });
    12 12 │   ({ 0b1: 1 });
  

```

```
useSimpleNumberKeys.js:11:4 lint/nursery/useSimpleNumberKeys  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Octal number literal is not allowed here.
  
     9 │ ({ 1n: 1 });
    10 │ ({ 0x1: 1 });
  > 11 │ ({ 012: 1 });
       │    ^^^
    12 │ ({ 0b1: 1 });
    13 │ ({ 0o1: 1 });
  
  i Safe fix: Replace 012 with 9
  
     9  9 │   ({ 1n: 1 });
    10 10 │   ({ 0x1: 1 });
    11    │ - ({·012:·1·});
       11 │ + ({·9:·1·});
    12 12 │   ({ 0b1: 1 });
    13 13 │   ({ 0o1: 1 });
  

```

```
useSimpleNumberKeys.js:12:4 lint/nursery/useSimpleNumberKeys  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Binary number literal in is not allowed here.
  
    10 │ ({ 0x1: 1 });
    11 │ ({ 012: 1 });
  > 12 │ ({ 0b1: 1 });
       │    ^^^
    13 │ ({ 0o1: 1 });
    14 │ ({ 1_0: 1 });
  
  i Safe fix: Replace 0b1 with 1
  
    10 10 │   ({ 0x1: 1 });
    11 11 │   ({ 012: 1 });
    12    │ - ({·0b1:·1·});
       12 │ + ({·1:·1·});
    13 13 │   ({ 0o1: 1 });
    14 14 │   ({ 1_0: 1 });
  

```

```
useSimpleNumberKeys.js:13:4 lint/nursery/useSimpleNumberKeys  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Octal number literal is not allowed here.
  
    11 │ ({ 012: 1 });
    12 │ ({ 0b1: 1 });
  > 13 │ ({ 0o1: 1 });
       │    ^^^
    14 │ ({ 1_0: 1 });
    15 │ ({ 0.1e1_2: "ed" });
  
  i Safe fix: Replace 0o1 with 1
  
    11 11 │   ({ 012: 1 });
    12 12 │   ({ 0b1: 1 });
    13    │ - ({·0o1:·1·});
       13 │ + ({·1:·1·});
    14 14 │   ({ 1_0: 1 });
    15 15 │   ({ 0.1e1_2: "ed" });
  

```

```
useSimpleNumberKeys.js:14:4 lint/nursery/useSimpleNumberKeys  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Number literal with underscore is not allowed here.
  
    12 │ ({ 0b1: 1 });
    13 │ ({ 0o1: 1 });
  > 14 │ ({ 1_0: 1 });
       │    ^^^
    15 │ ({ 0.1e1_2: "ed" });
    16 │ ({ 11_1.11: "ee" });
  
  i Safe fix: Replace 1_0 with 10
  
    12 12 │   ({ 0b1: 1 });
    13 13 │   ({ 0o1: 1 });
    14    │ - ({·1_0:·1·});
       14 │ + ({·10:·1·});
    15 15 │   ({ 0.1e1_2: "ed" });
    16 16 │   ({ 11_1.11: "ee" });
  

```

```
useSimpleNumberKeys.js:15:4 lint/nursery/useSimpleNumberKeys  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Number literal with underscore is not allowed.
  
    13 │ ({ 0o1: 1 });
    14 │ ({ 1_0: 1 });
  > 15 │ ({ 0.1e1_2: "ed" });
       │    ^^^^^^^
    16 │ ({ 11_1.11: "ee" });
    17 │ 
  
  i Safe fix: Replace 0.1e1_2 with .1e12
  
    13 13 │   ({ 0o1: 1 });
    14 14 │   ({ 1_0: 1 });
    15    │ - ({·0.1e1_2:·"ed"·});
       15 │ + ({·.1e12:·"ed"·});
    16 16 │   ({ 11_1.11: "ee" });
    17 17 │   
  

```

```
useSimpleNumberKeys.js:16:4 lint/nursery/useSimpleNumberKeys  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Number literal with underscore is not allowed.
  
    14 │ ({ 1_0: 1 });
    15 │ ({ 0.1e1_2: "ed" });
  > 16 │ ({ 11_1.11: "ee" });
       │    ^^^^^^^
    17 │ 
  
  i Safe fix: Replace 11_1.11 with 111.11
  
    14 14 │   ({ 1_0: 1 });
    15 15 │   ({ 0.1e1_2: "ed" });
    16    │ - ({·11_1.11:·"ee"·});
       16 │ + ({·111.11:·"ee"·});
    17 17 │   
  

```


